alonzo/impl/src/Cardano/Ledger/Alonzo/TxBody.hs

```diff
@@ -324,6 +324,50 @@ pattern TxBody
 
 instance (c ~ Crypto era, Era era) => HashAnnotated (TxBody era) EraIndependentTxBody c
 
+-- We define these accessor functions manually, because if we define them using
+-- the record syntax in the TxBody pattern, they inherit the (AlonzoBody era)
+-- constraint as a precondition. This is unnecessary, as one can see below
+-- they need not be constrained at all. This should be fixed in the GHC compiler.
+
+txinputs :: TxBody era -> Set (TxIn (Crypto era))
+txinputs_fee :: TxBody era -> Set (TxIn (Crypto era))
+txouts :: TxBody era -> StrictSeq (TxOut era)
+txcerts :: TxBody era -> StrictSeq (DCert (Crypto era))
+txfee :: TxBody era -> Coin
+txwdrls :: TxBody era -> Wdrl (Crypto era)
+txvldt :: TxBody era -> ValidityInterval
+txUpdates :: TxBody era -> StrictMaybe (Update era)
+txADhash :: TxBody era -> StrictMaybe (AuxiliaryDataHash (Crypto era))
+txmint :: TxBody era -> Value (Crypto era)
+txExunits :: TxBody era -> ExUnits
```

Comment:
we might want to replace this with a function that add up all the per-script exunits instead (to return this per-transaciton value)